src/js/track.js

```diff
@@ -0,0 +1,107 @@
+var _ = require('underscore');
+var d3 = require('d3');
+var $ = require('jquery');
+var D3SVGCellRenderer = require('./D3SVGCellRenderer');
+var ReadOnlyObject = require('./ReadOnlyObject');
+var utils = require('./utils');
+
+var defaultTrackConfig = {
+	label: 'Gene',
+	datum_id: function(d) { return d['sample'];},
+	cell_height: 20,
+	track_height: 20,
+	track_padding: 2.5,
+}; 
+
+function Track(name, data, config, oncoprint_config) {
+	var self = this;
+	self.name = name;
+	self.config = $.extend({}, defaultTrackConfig, config || {}); // inherit from default
+	self.oncoprint_config = oncoprint_config;
+	
+	self.data = data;
+	var cell_renderer;
+
+	if (self.oncoprint_config.get('render') === 'table') {
+		cell_renderer = new D3SVGCellRenderer(self.data, self.oncoprint_config.extend(self.config));
+		cell_renderer.bindEvents(self);
+		self.renderer = new TrackTableRenderer(cell_renderer);
+	}
+	self.renderer.bindEvents(self);
+
+	self.bindEvents = function(oncoprint) {
+		var pass_down_from_oncoprint = ['sort.oncoprint', 'set_cell_width.oncoprint', 'set_cell_padding.oncoprint'];
+		_.each(pass_down_from_oncoprint, function(evt) {
+			$(oncoprint).on(evt, function(e, data) {
+				$(self).trigger(evt, data);
+			})
+		});
+		var pass_up_from_cell_renderer = ['cell_click.oncoprint', 'cell_mouseenter.oncoprint', 'cell_mouseleave.oncoprint'];
+		_.each(pass_up_from_cell_renderer, function(evt) {
+			$(cell_renderer).on(evt, function(e, data) {
+				$(oncoprint).trigger(evt, $.extend({}, data, {track: self}));
+			})
+		});
+	};
+
+	self.getLabel = function() {
+		// TODO: label decorations
+		return self.config.label;
+	};
+	
+	self.getDatumIds = function(sort_cmp) {
+		return _.map((sort_cmp && utils.stableSort(self.data, sort_cmp)) || self.data, 
+				self.config.datum_id
+				);
+	};
+
+	self.useRenderTemplate = function(templName, params) {
+		self.renderer.useTemplate(templName, params);
+	};
+
+	$(self).trigger('init.track.oncoprint', {label_text: self.getLabel()});
+}
+
+function TrackTableRenderer(cell_renderer) {
+	// coupled with OncoprintTableRenderer
```

Comment:
should probably put them together in a single file that oncoprint.js and track.js import from to emphasize this.